subprojects/testing-base/src/main/java/org/gradle/api/tasks/testing/AbstractTestTask.java

```diff
@@ -167,9 +170,14 @@ protected ListenerManager getListenerManager() {
         return testListenerInternalBroadcaster;
     }
 
-    /**
-     * ATM. for testing only
-     */
+    @Internal
+    @VisibleForTesting
+    void setTestListenerInternal(TestListenerInternal testListenerInternal) {
```

Comment:
Instead of adding a setter, you could use subclassing in the unit test. That would make the production code simpler to read.

Reply:
done.